### PR TITLE
Add YANG parser

### PR DESCRIPTION
**Pull request by Hubro**
I spent the last couple of weeks learning about Tree-sitter and implemented a parser for YANG, which I use extensively at work. It would be cool to add it to nvim-treesitter. I made a highlights query as well, it works really well:

image

### CODE REVIEW

File: queries/yang/highlights.scm [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+(built_in_type) @type.builtin
+(integer) @number
+(boolean) @boolean
+(date) @number
+(range) @string.escape
+(unquoted_range) @string.escape
+(yang_version) @string.escape
+(identifier) @variable
+(node_identifier) @variable
+(glob) @string
```
Comment (vigoux) [inline queries/yang/highlights.scm:21]: Is this node representing a regex?
If it is, it might be worth adding an injection for those nodes.

Comment (Hubro) [reply queries/yang/highlights.scm:21]: No, glob just matches a literal `*`, but there is one YANG statement called `pattern` that does take a regex. However, currently it's just parsed as a generic statement just like everything else, and its argument is parsed as a string. I'll note it down as a future improvement.

Comment (stsewd) [reply queries/yang/highlights.scm:21]: we have a `string.regex` hl group if you wish to add it.

**Review by stsewd — APPROVED** [review]
Thanks!

**Comment by stsewd** [comment]
He, cases like this the bot merge too soon p: (leaving an optional suggestion for the author)

**Pull request merged**